### ISSUE TITLE
HUD inaccuracies

### ISSUE THREAD

**Issue by Diamond-Hero** [Fluvian/tpp#82]
I've noticed inaccuracies with every number and icon from the HUD (excluding the X). The most notable inaccuracies: the Mario icon has most of the details wrong, as well as the coin missing some details. Every one of the numbers has either missing details, incorrect shaping, or incorrect details. However, I can't really speak on their color, so I trust that's fine. I propose these improvements to these HUD elements, as shown in the attached image. I will give further details on what I changed and why I changed it if requested.
![HUD](https://user-images.githubusercontent.com/74314472/180632303-4b530a63-009a-4ece-bf18-e814f5409e84.png)
![REFS](https://user-images.githubusercontent.com/74314472/180632305-831ff142-1c32-44b1-a805-13b22de0858b.png)

**Comment by iProgramMC** [comment]
![betahDDD](https://user-images.githubusercontent.com/17717494/180645490-e16d311e-6483-409f-973d-795a6d06a419.png)
This is the best reference I have in regards to the first few number icons

**Comment by EmmerNerd** [comment]
A lot of these edits represent extremely sharpened, artifacted versions of the icons.

TPP's icon sprites were modified directly to account for scan artifacting, while your "edits" add artifacts which were intentionally removed.

You also shouldn't be heavily sharpening your references so much, it emphasizes existing artifacts and adds more.

**Comment by Diamond-Hero** [comment]
Hmm, I see. Does this work better? I removed certain details that might be a product of scan artifacting, toned down some of the shading so its not so sharpened, and did all this while retaining the details I added that are present in every reference I have.
![New HUD](https://user-images.githubusercontent.com/74314472/180650224-e2d58379-8f00-4cba-80a0-43b9ff69b589.png)

**Comment by EmmerNerd** [comment]
I still think that the HUD as it stands, is mostly accurate.

If necessary changes are needed I'm sure Lurondor will properly edit the HUD.

**Comment by Diamond-Hero** [comment]
Fair enough, I don't have to make the changes. But things like the 4, 7, 9, and 1 do need some changes to its shape. From the four 7 refs I have, its undeniable that its shape is wonky in the actual ssk build (the "back" is not a straight as in TPPs). It's already been mentioned before that the 1 is actually nine pixels wide instead of eight. The right "stub" of the 4 is shaped incorrectly as well. It's not square, its rounded. And in a few of the references it shows some pixels that show different shading for the "stub" as well. And lastly, the bottom of the 9 is not as rounded as TPPs. The bottom right of TPPs 9 is the same as the bottom right of the 0, and in the same place. Here's a screenshot of a 0 overlayed over a screenshot of a 9, both from the same video footage. As you can see, the 9 does not have the same bottom right as the 0, so the actual shape is how I edited it in my improvements. 
![0 compared to 9](https://user-images.githubusercontent.com/74314472/180651350-8a4e4170-02a2-4858-97cc-c9292a70e231.gif)
Also, in neither screenshots nor video do I see this shading pixel here.
![image](https://user-images.githubusercontent.com/74314472/180651414-1fd0dc36-9398-4502-bd06-c968ede59517.png)

**Comment by IForgorskull** [comment]
shapes of the numbers are accurate 100%, expect the 1.

But the colors & shading are absolutely wrong for most of them

**Comment by EmmerNerd** [comment]
@IForgorskull No, they're not.

You're still basing your view of these numbers on awful magazine scans.
I understand that people are trying to help with TPP, but please stop picking for issues that aren't there.

**Comment by Fluvian** [comment]
@IForgorskull 

And your *concrete* proof is...?

**Comment by iProgramMC** [comment]
I believe they mean that the 1 is 1 pixel longer. It may not be the case, but the rightmost column of the 1 matches slightly differently

![image](https://user-images.githubusercontent.com/17717494/181264583-76cda897-7df8-4a73-aa58-3777970002f3.png)
![image](https://user-images.githubusercontent.com/17717494/181264851-8d90c638-c881-4bbe-8433-0f33285c68b4.png)

**Comment by EmmerNerd** [comment]
The proof is that we spent over 2 years on TPP.
If these inaccuracies were legit, they would have been fixed by now.

The way TPP does things is snuffing out artifacting manually to get a result which looks correct under the same circumstances.

A lot of shading and shape critiques account for subpixelling, colour blending, and artifacting.
These are intentionally avoided because they weren't there.

Obviously magazine scans are the best we have, but you can't use them note-for-note.

**Comment by Diamond-Hero** [comment]
I understand that. Definitely those are factors to take in, but if there is a detail that is present in literally every reference we have, there is no doubt in my mind it is actually there. Because the scans/videos show it I feel like there's no denying it. "If these issues were present, Lurondor would likely have fixed them by now" is not good enough proof at all that the current icons/numbers are 100% accurate. I know you guys have been working on this for years, but that doesn't mean mistakes have not been made, even tiny pixel ones. I don't see proof that what was done with the HUD is 100% accurate, as even in its current state there are oddities such as hardly noticeable off-white pixels in random areas or other shading pixels and such that don't make sense because there is no proof of them existing. 

You've probably noticed that I've been slowly updating my edits to be more and more accurate. I've been looking into the numbers a little more, and as I go along the more I realize that TPP's HUD _is_ mostly accurate, but not _entirely_ accurate. I can't just trust that what has been done is an accurate job when I can see the inaccuracies with my own eyes. Though yes what IForgorskull said was incorrect (except the fact that the 1 is the wrong shape), that doesn't mean there aren't issues at all, just not the ones they said. 

Since I opened up this issue, I've been working on a document detailing my logic and proof for every issue I have found with the HUD, but I no longer see the point in continuing through all that work if it's just going to be ignored because the HUD "would've been fixed by now". I'm not gonna pretend I can accurately identify whether something is caused by artifacting, color blending, or subpixelling, but I can at least list all the issues I think are present and others that know more about that than me can help identify what is an inaccuracy or what's not through proof and logic/reasoning. My endgoal in this is just to make this project the best it can be.

**Comment by Fluvian** [comment]
@Diamond-Hero Instead of complaining about what other people say, why not post that list of proof in the first place? I wouldn't accept anything without concrete proof that the changes made were correct anyway.

**Comment by Diamond-Hero** [comment]
By in the first place do you mean right off the bat when opening up an issue?

**Comment by Fluvian** [comment]
@Diamond-Hero 

Yes, one could expect that from a project aimed at being accurate.
So far what was posted was an asset where it wasn't clear what was changed and without any proof if the changes were even correct.

**Comment by iProgramMC** [comment]
I think the "Lurondor would have fixed the issues by now" argument is fallacious.  It can take any amount of time for things to be noticed, for instance, the first unused textures in SM64 were found by VL-Tone in ~2005, 11 years after the game's release, and people had dumped the ROM for a long time.

**Comment by EmmerNerd** [comment]
> 

Literally not long before the release was planned out, lmao.
Bare in mind that unlike VL-Tone, we have been looking at a bunch of Spaceworld materials for longer than most have even been part of the community.

The reason these things are set in stone is because we have spent enough time on them to have faith.
Spaceworld research has existed since before the game even came out.

I'm mostly upset because our work is being undermined as inaccurate when a lot of the counter arguments are the things we specifically looked into while creating these things.

The argument of "the issues were known" is foolish to bring up, because they are known and were looked into in the creation and refining process of the HUD.

**Comment by iProgramMC** [comment]
"Spaceworld research has existed since before the game even came out."
I mean I guess people were looking in awe at the early version of SM64, but the research really wasn't that in-depth. The most in-depth they got was when a French magazine was analyzing some the differences between the alpha versions used in the B-roll video and the SpaceWorld demo.

**Comment by EmmerNerd** [comment]
It was indepth enough that people just before and not long after release discussed it.
Knuckles' Chaotix has been a dumped game for years and people still discover things, yes.
But there's a point where there isn't anything left to find.

We spent time making sure TPP is as close to the actual proto material as possible, it was a project done out of passion and again, if these issues were verifiable, they'd have been patched out.

**Comment by EmmerNerd** [comment]
I don't like to repeat myself on this, but it's something that we prioritised getting right.
It was one of the few things consistently being looked into even in late dev, and the HUD sprites themselves are still mostly accurate.

There are small tweaks that could be made, but the shaping and shading on things like the numbers is not part of that.

Mario's icon and the Coin are the only two which necessarily need proper edits.

**Comment by EmmerNerd** [comment]
Things like the shape of the 1 icon are because of a mix of a dozen aspects.
N64's image filtering, mixed with primitive screen capturing methods, mixed with colour blending CAUSED by that, will drastically alter even the most clean screenshot you could possibly get.

You can never completely trust a scan, and ironically there are places where it's better to trust a video than an image.

**Comment by EmmerNerd** [comment]
Again, I've given examples of other games where the same process has been applied to sprite art, and then the real thing looks entirely different.

You can NEVER trust even a high quality screencap in a magazine, because there are many phases of degradation the image will go through.

Resizing it to fit into the page also takes a toll on detail, and you can't restore any of that lost detail.

**Comment by iProgramMC** [comment]
I agree with your point on magazines. The scans tend to have degraded quality. Ink on a sheet of paper tends to not be as precise as can be, and printers in the 1990s weren't great. Are you able to explain in what cases video can be more accurate to the real thing than image scans from paper?

**Comment by Fluvian** [comment]
Bottom line is, we don't know what colors the HUD actually had, it's just best guesses, the shading is part of that too, the luminosity of each pixel is not exactly known either simply because too much information was lost.

It's like yellowed plastics on old computers where you don't know what color it originally had other than it being less yellow.

**Comment by iProgramMC** [comment]
On another note, I've looked at a video featuring TPP itself (this one: https://www.youtube.com/watch?v=rLScqZyB16Q) and the Mario icon looks pretty darn similar to scans despite not actually looking like that:

![image](https://user-images.githubusercontent.com/17717494/181494537-9e9d1045-c36d-4352-a10a-793c8749d7c9.png)

**Comment by IForgorskull** [comment]
💀

**Comment by IForgorskull** [comment]
> I understand that. Definitely those are factors to take in, but if there is a detail that is present in literally every reference we have, there is no doubt in my mind it is actually there. Because the scans/videos show it I feel like there's no denying it. "If these issues were present, Lurondor would likely have fixed them by now" is not good enough proof at all that the current icons/numbers are 100% accurate. I know you guys have been working on this for years, but that doesn't mean mistakes have not been made, even tiny pixel ones. I don't see proof that what was done with the HUD is 100% accurate, as even in its current state there are oddities such as hardly noticeable off-white pixels in random areas or other shading pixels and such that don't make sense because there is no proof of them existing.
> 
> You've probably noticed that I've been slowly updating my edits to be more and more accurate. I've been looking into the numbers a little more, and as I go along the more I realize that TPP's HUD _is_ mostly accurate, but not _entirely_ accurate. I can't just trust that what has been done is an accurate job when I can see the inaccuracies with my own eyes. Though yes what IForgorskull said was incorrect (except the fact that the 1 is the wrong shape), that doesn't mean there aren't issues at all, just not the ones they said.
> 
> Since I opened up this issue, I've been working on a document detailing my logic and proof for every issue I have found with the HUD, but I no longer see the point in continuing through all that work if it's just going to be ignored because the HUD "would've been fixed by now". I'm not gonna pretend I can accurately identify whether something is caused by artifacting, color blending, or subpixelling, but I can at least list all the issues I think are present and others that know more about that than me can help identify what is an inaccuracy or what's not through proof and logic/reasoning. My endgoal in this is just to make this project the best it can be.

devs be like "since we worked many many years into this, we are 100% sure what's wrong and right, therefore whenever someone report something is wrong, we get into defense mode!"

Contributing on this seems pointless if this is what happens...

**Comment by iProgramMC** [comment]
They seem to be quite overly confident in terms of their HUD. I still think the "Lurondor would have fixed the issues by now" argument is fallacious.

As EmmerNerd pointed out, 
> You can NEVER trust even a high quality screencap in a magazine, because there are many phases of degradation the image will go through.

Which (from what I understand) means that we will never know if the HUD is truly accurate to the source material.

**Comment by Diamond-Hero** [comment]
I don't think these replies were necessary? It seems pointless to continue arguing this same point when I don't even have all of my proof gathered and presented.

**Comment by IForgorskull** [comment]
> They seem to be quite overly confident in terms of their HUD. I still think the "Lurondor would have fixed the issues by now" argument is fallacious.
> 
> As EmmerNerd pointed out,
> 
> > You can NEVER trust even a high quality screencap in a magazine, because there are many phases of degradation the image will go through.
> 
> Which (from what I understand) means that we will never know if the HUD is truly accurate to the source material.

100% accuracy isn't possible with only scans (atleast for Sprites)

**Comment by iProgramMC** [comment]
You're right. I'm going to eagerly but patiently wait for your promised document.

**Comment by Fluvian** [comment]
> > I understand that. Definitely those are factors to take in, but if there is a detail that is present in literally every reference we have, there is no doubt in my mind it is actually there. Because the scans/videos show it I feel like there's no denying it. "If these issues were present, Lurondor would likely have fixed them by now" is not good enough proof at all that the current icons/numbers are 100% accurate. I know you guys have been working on this for years, but that doesn't mean mistakes have not been made, even tiny pixel ones. I don't see proof that what was done with the HUD is 100% accurate, as even in its current state there are oddities such as hardly noticeable off-white pixels in random areas or other shading pixels and such that don't make sense because there is no proof of them existing.
> > You've probably noticed that I've been slowly updating my edits to be more and more accurate. I've been looking into the numbers a little more, and as I go along the more I realize that TPP's HUD _is_ mostly accurate, but not _entirely_ accurate. I can't just trust that what has been done is an accurate job when I can see the inaccuracies with my own eyes. Though yes what IForgorskull said was incorrect (except the fact that the 1 is the wrong shape), that doesn't mean there aren't issues at all, just not the ones they said.
> > Since I opened up this issue, I've been working on a document detailing my logic and proof for every issue I have found with the HUD, but I no longer see the point in continuing through all that work if it's just going to be ignored because the HUD "would've been fixed by now". I'm not gonna pretend I can accurately identify whether something is caused by artifacting, color blending, or subpixelling, but I can at least list all the issues I think are present and others that know more about that than me can help identify what is an inaccuracy or what's not through proof and logic/reasoning. My endgoal in this is just to make this project the best it can be.
> 
> devs be like "since we worked many many years into this, we are 100% sure what's wrong and right, therefore whenever someone report something is wrong, we get into defense mode!"
> 
> Contributing on this seems pointless if this is what happens...

Honestly you're not helping the situation with a snarky comment like this, all you've said was that the shading/coloring was *absolutely* wrong and provided no evidence.

**Comment by Supergub5871** [comment]
idk what to say but here a zip having sm64 beta hud btw its not the ones from the real beta
https://drive.google.com/file/d/1HuXG9SlUQ_9-jPQ8EyV0NR_GreAY2q71/view?usp=sharing

**Comment by iProgramMC** [comment]
What is the point of posting that?

**Comment by Supergub5871** [comment]
nothing

**Comment by Supergub5871** [comment]
just to help i think

**Comment by Supergub5871** [comment]
but here a screenshot using it
![GLideN64_SUPER_MARIO_64_000](https://user-images.githubusercontent.com/90076568/183308593-02777657-4760-4b3e-b59a-409613892f93.png)

**Comment by iProgramMC** [comment]
Please do not spam multiple issue comments, keep it to one per reply. This isn't Discord.
Also, there's already a perfectly fine HUD in TPP, I'm not sure they need to use yours (or whoever else you stole it from)

**Comment by Supergub5871** [comment]
sorry 

am stupid

**Comment by Diamond-Hero** [comment]
It looks like they took the most recent improvements that I posted, made them separate images, and put them into a zip. I think the point was to put my improvements into a useable format for them? But I'm not even done with them yet, and I also haven't finished the document yet as I've said before, which when done would start discussion on what is fine with the improvements and what needs to be changed. Though I don't know when I will finish the doc, as I have been working on it bit by bit. Not even close to being done as of right now.

**Comment by TorutheRedFox** [comment]
ignoring magazine scans isn't the best idea either because based on one I've figured out that Mario's shading was different than final long before the gigaleak, but couldn't put my finger on exactly how

it turned out that the secondary (shadow) colors were half as bright as in the final

**Comment by wijayodustin021** [comment]
mine

**Comment by plpep69** [comment]
> ignoring magazine scans isn't the best idea either because based on one I've figured out that Mario's shading was different than final long before the gigaleak, but couldn't put my finger on exactly how
> 
> it turned out that the secondary (shadow) colors were half as bright as in the final

is beacuse the tpp team is stuped

**Comment by plpep69** [comment]
> They are probably a million times smarter than you are, kid
> 
> > > ignoring magazine scans isn't the best idea either because based on one I've figured out that Mario's shading was different than final long before the gigaleak, but couldn't put my finger on exactly how
> > > it turned out that the secondary (shadow) colors were half as bright as in the final
> > 
> > 
> > is beacuse the tpp team is stuped


https://user-images.githubusercontent.com/110211838/191352066-f3a5a4f8-c2a4-4729-832e-43a108febc11.mp4

**Comment by vakhdmyt06** [comment]
> > They are probably a million times smarter than you are, kid
> > > > ignoring magazine scans isn't the best idea either because based on one I've figured out that Mario's shading was different than final long before the gigaleak, but couldn't put my finger on exactly how
> > > > it turned out that the secondary (shadow) colors were half as bright as in the final
> > > 
> > > 
> > > is beacuse the tpp team is stuped
> 
>  b290e376258d05aabf3c33dac0dd8632.mp4

i never said im smart too
